From e499a4190e343618ffda8ac04148f52906e14e44 Mon Sep 17 00:00:00 2001
From: pcpa <paulo.cesar.pereira.de.andrade@gmail.com>
Date: Sat, 27 Sep 2014 16:04:16 -0300
Subject: [PATCH] su: print proper core dump message when killed

An example is, in one terminal "sudo su -; echo $$", and in
another terminal, "kill -9  $PID" (the pid of the su -). It
should not print "(core dumped)", unless the kill signal
specified so, e.g. kill -7 or kill -11.

Signed-off-by: pcpa <paulo.cesar.pereira.de.andrade@gmail.com>

Upstream-commit: 1d966345df6721dc1da257867a18a84f1d9f98d5
Signed-off-by: Kamil Dudka <kdudka@redhat.com>
---
 src/su.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/su.c b/src/su.c
index 47c7364..d0d89ff 100644
--- a/src/su.c
+++ b/src/su.c
@@ -618,10 +618,9 @@ run_shell (char const *shell, char const *command, char **additional_args,
       {
         if (WIFSIGNALED (status))
           {
+            fprintf (stderr, "%s%s\n", strsignal (WTERMSIG (status)),
+                     WCOREDUMP (status) ? _(" (core dumped)") : "");
             status = WTERMSIG (status) + 128;
-            if (WCOREDUMP (status))
-              fprintf (stderr, _("%s (core dumped)\n"),
-                 strsignal (WTERMSIG (status)));
           }
         else
           status = WEXITSTATUS (status);
-- 
2.5.2

